server/topic.go

```diff
@@ -1350,10 +1333,12 @@ func (t *Topic) thisUserSub(sess *Session, pkt *ClientComMessage, asUid types.Ui
 				userData.modeWant = (modeWant & types.ModeCSys) | types.ModeWrite | types.ModeJoin
 			}
 		} else if asChan {
```

Comment:
Question regarding line 1400/1385 down below:
do we want `private` in `userData` for channel readers?

Reply:
In general we do need it: it's used to store `archived` status. Do we need to cache it for the duration of the session? Maybe not. Let me look into it.